src/main/java/org/testng/TestNG.java

```diff
@@ -294,7 +295,16 @@ private void parseSuite(String suitePath) {
         s.setThreadCount(this.m_threadCount);
         // If test names were specified, only run these test names
         if (m_testNames != null) {
-          m_suites.add(XmlSuiteUtils.cloneIfContainsTestsWithNamesMatchingAny(s, m_testNames));
+          XmlSuiteUtils.cloneIfContainsTestsWithNamesMatchingAny(s, m_testNames);
+          List<String> missMatchedTestname = XmlSuiteUtils.getMissMatchedTestNames(m_testNames);
+          if (CollectionUtils.hasElements(missMatchedTestname)) {
+            XmlSuiteUtils.resetField();
+            throw new TestNGException("The test(s) <" + Arrays.toString(missMatchedTestname.toArray())
```

Comment:
I think we can remove the `Arrays.toString()` call here. The `List.toString()` itself should be sufficient.

Reply:
No, we can't.  java.util.List.toArray() doesn't override Object::toString method. So if remvoe Arrays.toString(), only name+hexadecimal would be displayed.

Reply:
@eyehwan - Sorry if my comment wasn't clear enough. 

I was stating that we could just use 

```java
throw new TestNGException("The test(s) <" + missMatchedTestname + "> cannot be found.");
```

instead of 

```java
throw new TestNGException("The test(s) <" + Arrays.toString(missMatchedTestname.toArray())+ "> cannot be found.");
```

The call to `Arrays.toString()` is totally not needed.